### PR TITLE
Split TC server-side plugins lib and api

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

File: clustered/integration-test/build.gradle [hunk 1]
```diff
@@ -66,6 +66,7 @@ task copyServerLibs(type: Copy) {
 test {
   dependsOn copyServerLibs
   environment 'JAVA_HOME', testJava.javaHome
+//  environment "JAVA_OPTS", "-verbose:class"
```
Comment (myronkscott) [inline clustered/integration-test/build.gradle]: keep it?

File: clustered/integration-test/src/test/java/org/ehcache/clustered/management/AbstractClusteringManagementTest.java [hunk 2]
```diff
@@ -333,11 +337,14 @@ public static void waitForAllNotifications(String... notificationTypes) throws I
         nmsService.waitForMessage(message -> {
           if (message.getType().equals("NOTIFICATION")) {
             for (ContextualNotification notification : message.unwrap(ContextualNotification.class)) {
-              if (waitingFor.remove(notification.getType())) {
+              if ("org.terracotta.management.entity.nms.client.NmsEntity".equals(notification.getContext().get("entityType"))) {
+                LOGGER.info("IGNORE:" + notification); // this is the passive NmsEntity, sometimes we catch it, sometimes not
+              } else if (waitingFor.remove(notification.getType())) {
                 existingOnes.add(notification);
-//                System.out.println("Remove " + notification.getType());
-//                System.out.println("Still waiting for: " + waitingFor);
```
Comment (myronkscott) [inline clustered/integration-test/src/test/java/org/ehcache/clustered/management/AbstractClusteringManagementTest.java]: keep it?

File: clustered/integration-test/build.gradle (hunk 1 above)
Comment (chrisdennis) [reply clustered/integration-test/build.gradle]: No... will remove.

**Comment by mathieucarbou** [comment]
@chrisdennis @AbfrmBlr @myronkscott : do you really wan to merge now ?

I would rather focus on:

1. getting https://github.com/Terracotta-OSS/terracotta-platform/pull/584 merge, which is ready we just need to make tests pass in Azure (matter of 1 day perhaps).

2. Updating this PR to be rebased on the paltform kit

Because sincerely with @saurabhagas we still have tremendous work to do in EE to be able to merge so we really do not want to loose time updating the kitting for each projects. So it would be nice to coordinate our work to not do things duplicate.

Once platform is done, we go back to EE. We didn't plan any work around the Ehcache kit and we do not have any time for that...

File: clustered/clustered-dist/build.gradle [hunk 3]
```diff
@@ -41,26 +41,40 @@ dependencies {
...
-    exclude group: 'org.terracotta', module: 'entity-common-api'
-    exclude group: 'org.terracotta', module: 'packaging-support'
-    exclude group: 'org.terracotta.internal', module: 'tc-config-parser'
+  serverApis "org.terracotta:statistics:$statisticVersion"
+  serverApis("org.terracotta.management:monitoring-service-api:$terracottaPlatformVersion") {
+    transitive = false
   }
+  //  this is here because monitoring-service-api depends on mnm-common.  if that can be unwound,
+  //  mnm-common can go into lib
+  serverApis "org.terracotta.management.dist:mnm-common:$terracottaPlatformVersion"
```
Comment (mathieucarbou) [inline clustered/clustered-dist/build.gradle]: i have removed all transitive mnm shading for the new kit.

**Pull request merged**